whois-api/src/main/java/net/ripe/db/whois/api/httpserver/FilteredSlf4jRequestLogWriter.java

```diff
@@ -0,0 +1,29 @@
+package net.ripe.db.whois.api.httpserver;
+
+import org.eclipse.jetty.server.Slf4jRequestLogWriter;
+
+import java.io.IOException;
+
+public class FilteredSlf4jRequestLogWriter extends Slf4jRequestLogWriter {
+    private final String keyToFilter;
+
+    public FilteredSlf4jRequestLogWriter(String keyToFilter) {
+        this.keyToFilter = keyToFilter;
+    }
+
+    @Override
+    public void write(String requestEntry) throws IOException {
+        String regexString;
+
+        if (keyToFilter != null && keyToFilter.equalsIgnoreCase("apikey")) {
+            // Replace with "FILTERED" but leave the last 3 characters if its API keys
+            regexString = "(?<=(?i)(key=))(.+?(?=\\S{3}\\s))".replace("key", keyToFilter);
```

Comment:
Can you extract the regex String into a static field Pattern? Otherwise a pattern is compiled every time the method is called

Reply:
done

Reply:
Thanks!